control/communication.py

```diff
@@ -29,11 +31,17 @@ def receive(self):
         type of data returned should be the same as the type that was sent by
         the other device.
 
-        :returns: original data -- This data has the type that was sent. It is
+        :returns:
+        None -- if the time_out time is exceeded
+
+        original data -- This data has the type that was sent. It is
                                    up to the user to determine the type and use
                                    it accordingly.
 
         """
+        #serial readline the data
         jsoned_data = self.ser.readline()
+        if not jsoned_data:
+            return None
         unjsoned_data = json.loads(jsoned_data)
```

Comment:
I think you meant to put the try except thing here for handling json errors, right now if we get bad data it will throw and exception. you should catch it and return ```None``` if we get bad data

Reply:
Atleast that's what you made it seem like this is supposed to do from the comment you got at the top of this pull request